agreement/service_test.go

```diff
@@ -404,7 +411,17 @@ func (n *testingNetwork) crown(prophets ...nodeID) {
 	defer n.mu.Unlock()
 	n.crownedNodes = make(map[nodeID]bool)
 	for i := 0; i < len(prophets); i++ {
-		n.crownedNodes[nodeID(i)] = true
+		n.crownedNodes[prophets[i]] = true
+	}
+}
+
+// Star topology with the given nodes at the center; to revert, call repairAll
+func (n *testingNetwork) makeRelays(relays ...nodeID) {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+	n.relayNodes = make(map[nodeID]bool)
+	for i := 0; i < len(relays); i++ {
```

Comment:
you're aware of the 
```
for relay := range relays {
...
```
syntax, right ?

Reply:
or, here, `for _, relay := range relays`, since we want the nodeID, not the index.
But we conflate the two all over this file. But yes, you have a point.